manifests/config.pp

```diff
@@ -9,6 +9,14 @@
           unless  => "test /etc/alternatives/java -ef '${java::use_java_alternative_path}'",
         }
       }
+      if $java::use_java_home != undef {
+        exec { 'java-home-environment':
+          path    => '/bin',
+          user    => 'root',
+          command => "echo JAVA_HOME=${$java::use_java_home} >> /etc/environment",
+          unless  => "grep -Fx \"JAVA_HOME=${$java::use_java_home}\" \"/etc/environment\"",
+        }
```

Comment:
Hmm, I think this is exactly what `file_line` from stdlib is for: https://github.com/puppetlabs/puppetlabs-stdlib#file_line